src/contexts/BasketContext.tsx

```diff
@@ -0,0 +1,215 @@
+import { createContext, FC, PropsWithChildren, useEffect, useState } from 'react';
+import { Meal, Restaurant } from '../utils/api/restaurantsService/restaurantsService';
+import { MealInBasket, basketService } from '../utils/api/basketService/basketService';
+
+type BasketContext = {
+    /**
+     * Indicates whether basket is empty
+     */
+    isEmpty: boolean;
+    /**
+     * Restaurant which meals are in basket
+     */
+    restaurant: Restaurant | null;
+    /**
+     * Meals in basket
+     */
+    meals: MealInBasket[];
+    /**
+     * Indicates whether basket is communicating with server
+     */
+    isLoading: boolean;
+    /**
+     * Error message returned from server
+     */
+    errorMessage: string;
+    /**
+     * Longest cooking time of all meals in basket
+     */
+    cookingTime: number;
+    /**
+     * Total price of all meals in basket
+     */
+    sum: number;
+    /**
+     * Quantity of meals in basket
+     */
+    total: number | null;
+    /**
+     * Add restaurant and meal to basket
+     */
+    addToBasket: (newRestaurant: Restaurant, newMeal: Meal) => void;
+    /**
+     * Increment quantity of meals by 1
+     */
+    addMeal: (meal: MealInBasket) => void;
+    /**
+     * Decrement quantity of meals by 1
+     */
+    deleteMeal: (meal: MealInBasket) => void;
+    /**
+     * Removes restaurant and all meals from basket
+     */
+    emptyBasket: () => void;
+};
+
+export const BasketContext = createContext<BasketContext>({
+    isEmpty: true,
+    restaurant: null,
+    meals: [],
+    isLoading: false,
+    errorMessage: '',
+    cookingTime: 0,
+    sum: 0,
+    total: null,
+    addToBasket: () => {},
+    addMeal: () => {},
+    deleteMeal: () => {},
+    emptyBasket: () => {},
+});
+
+export const BasketProvider: FC<PropsWithChildren> = ({ children }) => {
+    const [restaurant, setRestaurant] = useState<Restaurant | null>(null);
+    const [meals, setMeals] = useState<MealInBasket[]>([]);
+    const [isLoading, setIsLoading] = useState(false);
+    const [errorMessage, setErrorMessage] = useState('');
+    const sum = meals.reduce((acc, current) => acc + current.price * current.quantity, 0);
+    const total = meals.filter((meal) => meal.quantity > 0).length;
+    const cookingTime = Math.max(...meals.map((meal) => meal.cookingTime));
+    const isEmpty = restaurant ? false : true;
+    const addMeal = async (newMeal: MealInBasket | Meal) => {
+        const isAlreadyInBasket = meals.find((meal: MealInBasket) => meal.id === newMeal.id);
+        if (isAlreadyInBasket) {
+            setIsLoading(true);
+            const res = await basketService.addMeal(newMeal);
+            if (res.status === 'error') {
+                setIsLoading(false);
+                setErrorMessage(res.error_message);
+            } else {
+                setIsLoading(false);
```

Comment:
```suggestion
            const res = await basketService.addMeal(newMeal);
            setIsLoading(false);
            if (res.status === 'error') {
                setErrorMessage(res.error_message);
            } else {
```
и в других местах так же

Reply:
починил